worksheet.rb

```diff
@@ -23,12 +46,242 @@
+            rating: 2,
+        },
+    ],
+
+    DR0002: [
+        {
+            date: "3rd Feb 2016",
+            cost: 25,
+            rider_id: :RD0073,
+            rating: 5,
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 15,
+            rider_id: :RD0013,
+            rating: 1,
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 35,
+            rider_id: :RD0066,
+            rating: 3,
+        },
+    ],
+
+    DR0003: [
+        {
+            date: "4th Feb 2016",
+            cost: 5,
+            rider_id: :RD0066,
+            rating: 5,
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 50,
+            rider_id: :RD0003,
+            rating: 2,
+        },
+    ],
+
+    DR0004: [
+        {
+            date: "3rd Feb 2016",
+            cost: 5,
+            rider_id: :RD0022,
+            rating: 5,
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 10,
+            rider_id: :RD0022,
+            rating: 4,
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 20,
+            rider_id: :RD0073,
+            rating: 5,
+        },
+    ],
+
+}
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# The number of rides each driver has given
+# The total amount of money each driver has made
+# The average rating for each driver
+# Which driver made the most money?
+# Which driver has the highest average rating?
+# For each driver, on which day did they make the most money?
+
+
+# Method gets number of rides a driver has given
+# Returns an integer
+def get_driver_total_rides(drivers, driver_id)
+  num_of_rides = drivers[driver_id].length
+end
+
+
+# Method gets total amount of money a driver has made
+# driver_id parameter is a symbol
+# Returns an integer
+def get_driver_earnings(drivers, driver_id)
+  total_earnings = 0
+
+  drivers[driver_id].each do |ride|
+    earnings_per_ride = ride[:cost]
+    total_earnings += earnings_per_ride
+  end
+
+  return total_earnings
+end
+
+
+# Method gets the average rating for a driver
+# driver_id parameter is a symbol
+# Returns a float
+def get_driver_average_rating(drivers, driver_id)
+  total_rating = 0.0
+
+  drivers[driver_id].each do |ride|
+    total_rating += ride[:rating]
+  end
+
+  num_of_rides = drivers[driver_id].length
+  average_rating = total_rating / num_of_rides
+
+  return '%.1f' % average_rating
+end
+
+
+# Method gets a driver's total earnings by date
+# Returns a hash {date: earnings}
+def get_driver_earnings_by_date(drivers, driver_id)
```

Comment:
Note that `get_driver_earnings`, `get_driver_average_rating` and `get_driver_earnings_by_date` all have the same `.each` loop. Consider combining this functionality into a single method and returning the information in a hash or array to reduce repeated code. It is a delicate balance between writing many methods so that they each only do one thing and combining functionality into a single method to reduce repeated code.